gravitee-am-service/src/main/java/io/gravitee/am/service/impl/TokenServiceImpl.java

```diff
@@ -113,7 +123,13 @@ public Completable deleteByUserId(String userId) {
                     LOGGER.error("An error occurs while trying to delete tokens by user {}", userId, ex);
                     return Completable.error(new TechnicalManagementException(
                             String.format("An error occurs while trying to find total tokens by user: %s", userId), ex));
-                });
+                })
+                .doOnEvent(error -> userService.findById(userId).doOnEvent((user, er) ->
+                                auditService.report(AuditBuilder.builder(ClientTokenAuditBuilder.class)
+                                        .tokenTarget(user)
```

Comment:
**thought**: do we have to log this token deletion here :thinking: ? Looks like this deleteByUserId is triggered on Logout and ResetPassword actions. Both of these actions have appropriate logs so maybe not useful as we do not know which tokens are revoked.

Reply:
I think it would be good to log it. I can add a revoked reason.